### ISSUE TITLE
Property tagName

### ISSUE THREAD

**Issue by meradoou** [alexdebril/feed-io#170]
Hello,
I get this error with https://feeds.feedblitz.com/scotch_io. There is no exception ?

```
PHP Notice:  Trying to get property 'tagName' of non-object in /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Atom.php on line 55
PHP Notice:  Trying to get property 'tagName' of non-object in /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Rss.php on line 72
PHP Notice:  Trying to get property 'tagName' of non-object in /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Rdf.php on line 43
PHP Fatal error:  Uncaught TypeError: strpos() expects parameter 1 to be string, null given in /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Rdf.php:43
Stack trace:
#0 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Rdf.php(43): strpos(NULL, 'rdf')
#1 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Reader.php(175): FeedIo\Standard\Rdf->canHandle(Object(FeedIo\Reader\Document))
#2 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Reader.php(161): FeedIo\Reader->getAccurateParser(Object(FeedIo\Reader\Document))
#3 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Reader.php(146): FeedIo\Reader->parseDocument(Object(FeedIo\Reader\Document), Object(FeedIo\Feed))
#4 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Reader.php(125): FeedIo\Reader->handleResponse(Object(FeedIo\Adapter\Guzzle\Response), Object(FeedIo\Feed))
#5 /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/FeedIo.php(302): FeedIo\Reader->read('https://feeds.f...', in /srv/http/vhosts/piso.local/vendor/debril/feed-io/src/FeedIo/Standard/Rdf.php on line 43
````

**Comment by alexdebril** [comment]
Hi,

It seems that feed-io acts like it doesn't recognize the feed's format, but when I read the feed using the CLI, I don't get any trouble : 

```sh
./bin/feedio read 'https://feeds.feedblitz.com/scotch_io&x=1'
```

Which version are you using ? Could you paste the code where you build the `FeedIo` instance before reading the feed please ?

**Comment by meradoou** [comment]
Hi,
I use `"debril/feed-io": "^4.2",`  and 

```
<?php
$feedIo = Factory::create()->getFeedIo();
$feed = $feedIo->read($url)->getFeed();
```

**Comment by meradoou** [comment]
$url = 'https://feeds.feedblitz.com/scotch_io';

**Comment by alexdebril** [comment]
the real URL of the feed is https://feeds.feedblitz.com/scotch_io&x=1, be careful of the argument at the end of the query string :-)

the real trouble here is the lack of feed discovery. If feed-io could discover feeds in HTML pages, it would have follow it and fetched the accurate XML stream.

this issue duplicates #157

**Comment by meradoou** [comment]
Thank you :)

**Comment by alexdebril** [comment]
you're welcome ! I close this issued as it is going to be resolved through #157
